### PR TITLE
fix(intake): map ATIF v1.7 invocation timing to span end times

### PR DESCRIPTION
**Pull request by BrianNewsom**
Implements two spec-compliance gaps in the ATIF ingest mapping:

- AtifToolCall now accepts the v1.7 `extra` field. NAT publishers write per-tool-call ancestry/invocation metadata there and the spec requires consumers to tolerate it; previously such payloads were rejected 422.
- Step, tool, and delegation spans now get end times. Explicit timing from the spec-blessed `extra["invocation"]` contract (AtifInvocationInfo, epoch-second start/end) is preferred; otherwise the end is inferred from the next timed step's start, with trailing steps ending at the verifier finish. A step's own timestamp is never used as its end and underivable ends stay NULL, so no zero-duration spans are fabricated. Ends that precede their span's start (out-of-order producers) are dropped.

Trajectory root windows now also account for invocation timestamps.

## Summary by CodeRabbit

* **New Features**
  * Added ATIF v1.7 support for `trajectory_id`, recursive `subagent_trajectories`, and per-tool-call `extra` metadata across the API/OpenAPI and CLI.
  * Embedded subagent trajectories are now expanded into the parent trace, bounded by a configurable max depth (requests beyond the limit are rejected).
* **Bug Fixes**
  * Improved nested span mapping and evaluator-result association for embedded trajectory trees.
  * Refreshed span timing to derive start/end from invocation windows and handle malformed or out-of-order timestamps safely.
* **Tests**
  * Added unit and integration coverage for v1.7 round-tripping, recursive expansion, depth enforcement, and timing edge cases.

### CODE REVIEW

**Comment by BrianNewsom** [comment]
@coderabbitai full review

**Comment by BrianNewsom** [comment]
@coderabbitai review

**Review by shanaiabuggy — APPROVED** [review]

**Comment by BrianNewsom** [comment]
@coderabbitai review

**Comment by BrianNewsom** [comment]
@coderabbitai review

**Comment by BrianNewsom** [comment]
@coderabbitai full review

File: services/intake/src/nmp/intake/spans/ingest/atif_mapping.py [hunk 1]
```diff
@@ -37,91 +38,175 @@
...
+                trajectory=subagent,
+                trace_session_id=trace_session_id,
+                trajectory_identity=(*trajectory_identity, "subagent", subagent.trajectory_id),
+                external_parent_span_id=trajectory_span.external_span_id,
                 ingested_at=ingested_at,
             )
-            for result_index, result in _observation_results_with_subagents(step)
-            for ref_index, subagent_ref in enumerate(result.subagent_trajectory_ref or [])
         )
     return spans
```
[the remark this answers is not in the document]
Comment (BrianNewsom) [reply services/intake/src/nmp/intake/spans/ingest/atif_mapping.py:217]: Addressed in bbf8d83c20. ATIF ingest now enforces configurable NMP_INTAKE_ATIF_MAX_SUBAGENT_DEPTH (default 64, hard maximum 256), uses iterative traversal in the helper walks, guards recursive span expansion, and returns HTTP 422 for an oversized tree. Unit and ClickHouse-backed integration regressions cover both the accepted boundary and rejection path.

File: services/intake/src/nmp/intake/spans/ingest/atif_mapping.py [hunk 2]
```diff
@@ -37,91 +38,175 @@
...
+                    spans.extend(
+                        _trajectory_tree_to_spans(
+                            workspace=workspace,
+                            trajectory=subagent,
+                            trace_session_id=trace_session_id,
+                            trajectory_identity=(*trajectory_identity, "subagent", subagent.trajectory_id),
+                            external_parent_span_id=step_span.external_span_id,
+                            ingested_at=ingested_at,
+                        )
+                    )
```
[the remark this answers is not in the document]
Comment (BrianNewsom) [reply services/intake/src/nmp/intake/spans/ingest/atif_mapping.py:180]: Addressed in bbf8d83c20. Referenced subagent trajectories now use the delegating step span start as their fallback; unreferenced embedded trajectories use the containing trajectory span start. The regression test verifies a timestamp-free nested child cannot start before its parent delegation.

**Comment by BrianNewsom** [comment]
@coderabbitai review

**Comment by BrianNewsom** [comment]
@coderabbitai review
